students/eowyn/session06/mailroom/test_mailroom.py

```diff
@@ -0,0 +1,122 @@
+              ('Margaret Atwood', 855, 2, 428)]
+    assert mailroom.generate_report_data() == output
+
+
+def test_setup_table():
+    ''' this is not working'''
+    output = """'Donor Name                              |
+    Total Given |Num Gifts |Average Gift'\n,
+    '----------------------------------------
+    -------------------------------------'"""
+    assert mailroom.setup_table() == output
+
+
+def test_setup_body():
+    ''' this is not working'''
+    list_data = mailroom.generate_report_data()
+    output = """'Heinz the Baron Krauss von Espy         $  3800.00        2      $  1900.00   \nFred Armisen                            $  1662.00
+3      $   554.00   \nMargaret Atwood                         $   855.00        2      $   428.00   \n'
+"""
+    assert mailroom.setup_body(list_data) == output
+
+def test_remove_inputquotes():
+    output = 'hello'
+    assert mailroom.remove_inputquotes("hello") == output
+
+
+
+def test_retrieve_donor_exists():
+    mailroom.DONORS = {'Fred Armisen': [240, 422, 1000],
+                       'Heinz the Baron Krauss von Espy': [1500, 2300],
+                       'Margaret Atwood': [300, 555]}
+    name = "Margaret Atwood"
+    output = [300, 555]
+    assert mailroom.retrieve_donations(name) == output
+
+def test_retrieve_donations_new():
+    mailroom.DONORS = {'Fred Armisen': [240, 422, 1000],
+                       'Heinz the Baron Krauss von Espy': [1500, 2300],
+                       'Margaret Atwood': [300, 555]}
+    name = "Paul Simon"
+    output = None
+    assert mailroom.retrieve_donations(name) == output
+
+
+def test_add_donation_new():
+    mailroom.DONORS = {'Fred Armisen': [240, 422, 1000],
+                       'Heinz the Baron Krauss von Espy': [1500, 2300],
+                       'Margaret Atwood': [300, 555]}
+    name = "Bob Mueller"
+    amount = 500
+    mailroom.add_donation(name, amount)
+    assert mailroom.retrieve_donations(name) == [amount]
+
+
+def test_add_donation_existing():
+    mailroom.DONORS = {'Fred Armisen': [240, 422, 1000],
+                       'Heinz the Baron Krauss von Espy': [1500, 2300],
+                       'Margaret Atwood': [300, 555]}
+    name = "Fred Armisen"
+    amount = 500
+    mailroom.add_donation(name, amount)
+    assert mailroom.retrieve_donations(name) == [240, 422, 1000, 500]
+
+
+def test_select_action1():
+    ''' not sure how to test if correct action was completed
+    something called "mock" has this capability?'''
+    pass
+
+
+def test_file_existance():
+    mailroom.DONORS = {'Fred Armisen': [240, 422, 1000],
+                       'Heinz the Baron Krauss von Espy': [1500, 2300],
+                       'Margaret Atwood': [300, 555]}
+    mailroom.send_letters()
+    c1 = os.path.isfile("Margaret_Atwood.txt")
+    c2 = os.path.isfile("Fred_Armisen.txt")
+    c3 = os.path.isfile("Heinz_the_Baron_Krauss_von_Espy.txt")
+    output = c1 and c2 and c3
+    assert output is True
```

Comment:
you can do this with multiple asserts:

assert os.path.isfile("Margaret_Atwood.txt")
assert os.path.isfile("Fred_Armisen.txt")
assert os.path.isfile("Heinz_the_Baron_Krauss_von_Espy.txt")

you could even put that in a loop to automate it

But be careful -- they could be left over from a previous test run!

don't worry about that for now though.